server/log_operation_manager.go

```diff
@@ -247,6 +251,42 @@ func (l *LogOperationManager) getLogIDs(ctx context.Context) ([]int64, error) {
 	return logIDs, nil
 }
 
+// logName maps a logID to a human-readable name, caching results along the way.
+// The human-readable name may non-unique so should only be used for diagnostics.
+func (l *LogOperationManager) logName(ctx context.Context, logID int64) string {
+	l.nameMutex.Lock()
+	defer l.nameMutex.Unlock()
+	if name, ok := l.logNames[logID]; ok {
+		return name
+	}
+	tx, err := l.info.Registry.AdminStorage.Snapshot(ctx)
+	if err != nil {
+		glog.Errorf("%v: failed to start transaction: %v", logID, err)
+		return "<err>"
+	}
+	defer tx.Close()
+	tree, err := tx.GetTree(ctx, logID)
+	if err != nil {
+		glog.Errorf("%v: failed to get log info: %v", logID, err)
+		return "<err>"
+	}
+	tx.Commit()
```

Comment:
You should probably verify that Commit's err is non-nil. Some DBs have reads that should only be trusted after a successful commit.

Reply:
Done.